ui/apps/dashboard/src/queries/useRerun.tsx

```diff
@@ -0,0 +1,58 @@
+import { Link } from '@inngest/components/Link';
+import { toast } from 'sonner';
+import { useMutation } from 'urql';
+
+import { graphql } from '@/gql';
+import { pathCreator } from '@/utils/urls';
+
+const mutation = graphql(`
+  mutation RerunFunctionRun($environmentID: ID!, $functionID: ID!, $functionRunID: ULID!) {
+    retryWorkflowRun(
+      input: { workspaceID: $environmentID, workflowID: $functionID }
+      workflowRunID: $functionRunID
+    ) {
+      id
+    }
+  }
+`);
+
+export function useRerun({
+  envID,
+  envSlug,
+  runID,
+}: {
+  envID: string;
+  envSlug: string;
+  runID: string;
+}) {
+  const [, mutate] = useMutation(mutation);
+
+  async function rerun({ fnID }: { fnID: string }): Promise<void> {
+    try {
+      const response = await mutate({
+        environmentID: envID,
+        functionID: fnID,
+        functionRunID: runID,
+      });
+      if (response.error) {
+        throw response.error;
+      }
+      const newRunID = response.data?.retryWorkflowRun?.id;
+      if (!newRunID) {
+        throw new Error('missing new run ID');
+      }
+
+      // Give user a link to the new run
+      toast.success(
```

Comment:
I think it should have a bit more text in the message before the link: "Successfully rerun." And then the "View rerun" link